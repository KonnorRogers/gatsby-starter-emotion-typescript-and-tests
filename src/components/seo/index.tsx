/**
 * SEO component that queries for data with
 *  Gatsby's useStaticQuery React hook
 *
 * See: https://www.gatsbyjs.org/docs/use-static-query/
 */

import React from "react";
import Helmet from "react-helmet";
import { useStaticQuery, graphql } from "gatsby";

interface Props {
  title: string;
  description?: string;
  lang?: string;
  meta?: [];
}

interface SiteProps {
  site: {
    siteMetadata: {
      title?: string;
      description?: string;
      author?: string;
    };
  };
}

export function PureSEO({
  site,
  description,
  lang,
  meta,
  title,
}: Props & SiteProps): JSX.Element {
  const metaDescription = description || site.siteMetadata.description;

  return (
    <Helmet
      htmlAttributes={{
        lang,
      }}
      title={title}
      titleTemplate={`%s | ${site.siteMetadata.title}`}
      meta={[
        {
          name: `description`,
          content: metaDescription,
        },
        {
          property: `og:title`,
          content: title,
        },
        {
          property: `og:description`,
          content: metaDescription,
        },
        {
          property: `og:type`,
          content: `website`,
        },
        {
          name: `twitter:card`,
          content: `summary`,
        },
        {
          name: `twitter:creator`,
          content: site.siteMetadata.author,
        },
        {
          name: `twitter:title`,
          content: title,
        },
        {
          name: `twitter:description`,
          content: metaDescription,
        },
      ].concat(meta || [])}
    />
  );
}

function SEO(attrs: Props): JSX.Element {
  const { site } = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
            description
            author
          }
        }
      }
    `
  );

  return <PureSEO site={site} {...attrs} />;
}

export default SEO;
