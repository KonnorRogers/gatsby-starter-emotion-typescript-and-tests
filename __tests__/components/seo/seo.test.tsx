import * as React from "react";
import { render, waitForDomChange } from "@testing-library/react";
import data from "~fixtures/seo-fixture";
import { PureSEO as SEO } from "~components/seo";

describe("SEO component", () => {
  test("Should properly render without errors", async () => {
    // https://github.com/testing-library/react-testing-library/issues/402 used to get this testing working
    render(<SEO title={data.title} site={data.site} />);
    await waitForDomChange();

    expect(document.querySelector("head")).toMatchSnapshot();
  });
});
