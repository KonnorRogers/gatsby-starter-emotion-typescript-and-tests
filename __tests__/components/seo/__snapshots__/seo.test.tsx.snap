// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO component Should properly render without errors 1`] = `
<head>
  <title>
    Test Title | Site Title
  </title>
  <meta
    content="My testing site"
    data-react-helmet="true"
    name="description"
  />
  <meta
    content="Test Title"
    data-react-helmet="true"
    property="og:title"
  />
  <meta
    content="My testing site"
    data-react-helmet="true"
    property="og:description"
  />
  <meta
    content="website"
    data-react-helmet="true"
    property="og:type"
  />
  <meta
    content="summary"
    data-react-helmet="true"
    name="twitter:card"
  />
  <meta
    content="Konnor Rogers"
    data-react-helmet="true"
    name="twitter:creator"
  />
  <meta
    content="Test Title"
    data-react-helmet="true"
    name="twitter:title"
  />
  <meta
    content="My testing site"
    data-react-helmet="true"
    name="twitter:description"
  />
</head>
`;
